### PR TITLE
Setup Azure DevOps CI 

### PR DESCRIPTION
**Pull request by ivan-nosar**
This PR adds `azure-pipelines.yml` file which would be used in Azure Pipelines in order to configure build definition for CI.
Tests uses the following matrix:
- macOS Mojave + Xcode 10.2.1
- macOS Mojave + Xcode 11

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  - script: |
+      sudo xcode-select --switch "/Applications/Xcode_$(XCODE_VERSION).app/Contents/Developer"
+    displayName: "Select Xcode $(XCODE_VERSION)"
+
+  - script: xcodebuild -version
+    displayName: "Version of xcodebuild"
+  
+  - script: git clone -b develop https://github.com/calabash/FBSimulatorControl.git ../FBSimulatorControl
+    displayName: Download FBSimulatorControl dependency
```
Comment (jmoody) [inline azure-pipelines.yml]: Let's not pull FBSimulatorControl.  The required .frameworks are committed to git, so let's use those.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  - script: |
+      git clone -b master https://$(GitHubAccessToken):x-oauth-basic@github.com/xamarinhq/calabash-codesign.git ../calabash-codesign
```
Comment (jmoody) [inline azure-pipelines.yml]: In multi-line scripts, always start with:

```
set -e
```

to force the script to exit if a subcommand failes.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      ../calabash-codesign/apple/create-keychain.sh
+    displayName: "Download and install keychain"
+
+  - script: bundle install
+    displayName: "Bundle install"
+
+  - script: make frameworks
+    displayName: "Make frameworks"
```
Comment (jmoody) [inline azure-pipelines.yml]: Remove this step.  See the note about FBSimulatorControl above.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  - script: make unit-tests
+    displayName: "Make unit-tests"
+
+  - script: make integration-tests
+    displayName: "Make integration-tests"
+
+  - script: make build
```
Comment (jmoody) [inline azure-pipelines.yml]: This should be run before the tests.

File: azure-pipelines.yml (hunk 2 above)
Comment (jmoody) [reply azure-pipelines.yml]: > ../calabash-codesign

I don't think we own that directory so we cannot put files there.

> $(GitHubAccessToken)

What GitHub Service Connection are you using?

File: azure-pipelines.yml (hunk 1 above)
Comment (ivan-nosar) [reply azure-pipelines.yml]: Done

File: azure-pipelines.yml (hunk 3 above)
Comment (ivan-nosar) [reply azure-pipelines.yml]: Done

File: azure-pipelines.yml (hunk 4 above)
Comment (ivan-nosar) [reply azure-pipelines.yml]: Done

File: azure-pipelines.yml (hunk 2 above)
Comment (ivan-nosar) [reply azure-pipelines.yml]: 1. Added `set -e`. Thanks!
2. Replaced to `./calabash-codesign/`
3. `$(GitHubAccessToken)` - is a secret variable defined in build definition. It's store private access token of Github user who have pull-access to the repository.

File: spec/resources.rb [hunk 5]
```diff
@@ -32,7 +32,13 @@ def self.exit_status(key)
 
   def self.shell(args)
     cmd = [Resources.instance.idm] + args
-    RunLoop::Shell.run_shell_command(cmd, {log_cmd: true, timeout: 180})
+
+    timeout = 180
+    if RunLoop::Environment.ci?
+      timeout = 300
```
Comment (jmoody) [inline spec/resources.rb]: This is really disappointing. :/

Comment (max-lobanov) [reply spec/resources.rb]: Most often, it passes very quick. But sometimes it takes more time for Xcode 11 (may be `beta` instability)

File: azure-pipelines.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  pool:
+    vmImage: $(IMAGE_POOL)
+
+  steps:
+
+  - script: sudo xcode-select --switch "/Applications/Xcode_$(XCODE_VERSION).app/Contents/Developer"
+    displayName: "Select Xcode $(XCODE_VERSION)"
+
+  - script: xcodebuild -version
+    displayName: "Version of xcodebuild"
```
Comment (jmoody) [inline azure-pipelines.yml]: Let's combine these two steps and add the same debugging information:

```
      set -e
	      sudo xcode-select --switch "/Applications/Xcode_$(XCODE_VERSION).app/Contents/Developer"
	      echo "Xcode version: $(xcrun xcodebuild -version)"
	      echo "Xcode path: $(xcrun -f xcodebuild)"
	      echo "Xcode select path: $(xcode-select --print-path)"
```

Comment (ivan-nosar) [reply azure-pipelines.yml]: Done

**Pull request merged**